### PR TITLE
[fmilib] Change to the github and update to fix bug of libexpat

### LINKED ISSUE #27890: [fmilib] Build error
**Host Environment**

- OS: Linux CentOS7
- Compiler: gcc (GCC) 8.3.1 20190311 (Red Hat 8.3.1-3)

**To Reproduce**
`./vcpkg install fmilab`

**Failure logs**
[config-x64-linux-dbg-err.log](https://github.com/microsoft/vcpkg/files/10039831/config-x64-linux-dbg-err.log)
[config-x64-linux-dbg-out.log](https://github.com/microsoft/vcpkg/files/10039832/config-x64-linux-dbg-out.log)
[vcpkg-manifest-install.log](https://github.com/microsoft/vcpkg/files/10039833/vcpkg-manifest-install.log)
[config-x64-linux-dbg-CMakeCache.txt.log](https://github.com/microsoft/vcpkg/files/10039834/config-x64-linux-dbg-CMakeCache.txt.log)

```
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:96 (message):
    Command failed: /home/vsm/vcpkg/downloads/tools/cmake-3.24.0-linux/cmake-3.24.0-linux-x86_64/bin/cmake /home/vsm/vcpkg/buildtrees/fmilib/src/ILibrary-2-bb711565e5.clean -G Ninja -DCMAKE_BUILD_TYPE=Debug -DCMAKE_INSTALL_PREFIX=/home/vsm/vcpkg/packages/fmilib_x64-linux/debug -DFMILIB_BUILD_TESTS=OFF -DFMILIB_BUILD_STATIC_LIB=ON -DFMILIB_BUILD_SHARED_LIB=OFF -DFMILIB_BUILDING_LIBRARY=OFF -DFMILIB_BUILD_WITH_STATIC_RTLIB=OFF -DCMAKE_MAKE_PROGRAM=/usr/bin/ninja -DCMAKE_SYSTEM_NAME=Linux -DBUILD_SHARED_LIBS=OFF -DVCPKG_CHAINLOAD_TOOLCHAIN_FILE=/home/vsm/vcpkg/scripts/toolchains/linux.cmake -DVCPKG_TARGET_TRIPLET=x64-linux -DVCPKG_SET_CHARSET_FLAG=ON -DVCPKG_PLATFORM_TOOLSET=external -DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON -DCMAKE_FIND_PACKAGE_NO_PACKAGE_REGISTRY=ON -DCMAKE_FIND_PACKAGE_NO_SYSTEM_PACKAGE_REGISTRY=ON -DCMAKE_INSTALL_SYSTEM_RUNTIME_LIBS_SKIP=TRUE -DCMAKE_VERBOSE_MAKEFILE=ON -DVCPKG_APPLOCAL_DEPS=OFF -DCMAKE_TOOLCHAIN_FILE=/home/vsm/vcpkg/scripts/buildsystems/vcpkg.cmake -DCMAKE_ERROR_ON_ABSOLUTE_INSTALL_DESTINATION=ON -DVCPKG_CXX_FLAGS= -DVCPKG_CXX_FLAGS_RELEASE= -DVCPKG_CXX_FLAGS_DEBUG= -DVCPKG_C_FLAGS= -DVCPKG_C_FLAGS_RELEASE= -DVCPKG_C_FLAGS_DEBUG= -DVCPKG_CRT_LINKAGE=dynamic -DVCPKG_LINKER_FLAGS= -DVCPKG_LINKER_FLAGS_RELEASE= -DVCPKG_LINKER_FLAGS_DEBUG= -DVCPKG_TARGET_ARCHITECTURE=x64 -DCMAKE_INSTALL_LIBDIR:STRING=lib -DCMAKE_INSTALL_BINDIR:STRING=bin -D_VCPKG_ROOT_DIR=/home/vsm/vcpkg -DZ_VCPKG_ROOT_DIR=/home/vsm/vcpkg -D_VCPKG_INSTALLED_DIR=/home/vsm/vsm/vsm-release2/vcpkg_installed -DVCPKG_MANIFEST_INSTALL=OFF -DFETCHCONTENT_FULLY_DISCONNECTED=ON
    Working Directory: /home/vsm/vcpkg/buildtrees/fmilib/x64-linux-dbg
    Error code: 1
    See logs for more information:
      /home/vsm/vcpkg/buildtrees/fmilib/config-x64-linux-dbg-CMakeCache.txt.log
      /home/vsm/vcpkg/buildtrees/fmilib/config-x64-linux-dbg-out.log
      /home/vsm/vcpkg/buildtrees/fmilib/config-x64-linux-dbg-err.log

Call Stack (most recent call first):
  scripts/cmake/vcpkg_configure_cmake.cmake:325 (vcpkg_execute_required_process)
  buildtrees/versioning_/versions/fmilib/9bf58761098e936528140d847911dab3613f5983/portfile.cmake:53 (vcpkg_configure_cmake)
  scripts/ports.cmake:147 (include)


error: building fmilib:x64-linux failed with: BUILD_FAILED

```

### PR DESCRIPTION
**Pull request by FrankXie05**
Fix https://github.com/microsoft/vcpkg/issues/27890

Note:   The official statement has been transferred to github,  https://jmodelica.org/ so it has been changed to `github` download. And the problem about `libexpat` has been solved in the latest version `2.4.1` https://github.com/modelon-community/fmi-library/issues/18#issuecomment-1247834281

- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [x] SHA512s are updated for each updated download
- [ ] ~~The "supports" clause reflects platforms that may be fixed by this new version~~
- [ ] ~~Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.~~
- [ ] ~~Any patches that are no longer applied are deleted from the port's directory.~~
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

File: ports/fmilib/vcpkg.json [hunk 1]
```diff
@@ -1,7 +1,6 @@
 {
   "name": "fmilib",
-  "version": "2.0.3",
-  "port-version": 5,
+  "version": "2.4.1",
   "description": "FMI library is intended as a foundation for applications interfacing FMUs (Functional Mockup Units) that follow FMI Standard. This version of the library supports FMI 1.0 and FMI2.0.",
   "homepage": "https://www.fmi-standard.org/",
   "license": null,
```
Comment (JonLiu1993) [inline ports/fmilib/vcpkg.json:6]:
```suggestion
  "license": "BSD-3-Clause",
```

Comment (JonLiu1993) [reply ports/fmilib/vcpkg.json:6]: https://github.com/modelon-community/fmi-library/blob/master/LICENSE.md

Comment (FrankXie05) [reply ports/fmilib/vcpkg.json:6]: The upstream does not quote the `BSD-3-Clause` license.

**Review by JonLiu1993 — DISMISSED** [review]

File: ports/fmilib/vcpkg.json (hunk 1 above)
Comment (dg0yt) [reply ports/fmilib/vcpkg.json:6]: Copy and paste to https://tools.spdx.org/app/check_license/. Or compare manually.

Comment (dg0yt) [reply ports/fmilib/vcpkg.json:6]: There is also https://github.com/modelon-community/fmi-library/blob/master/FMILIB_License.txt.

**Comment by dg0yt** [comment]
Maybe fixing the merging of staticlib is better achieved with recent CMake's WHOLEARCHIVE:
https://cmake.org/cmake/help/v3.25/release/3.24.html?highlight=whole_archive#generator-expressions

**Review by JavierMatosD — APPROVED** [review]

**Comment by dg0yt** [comment]
This PR didn't fix #27890.

**Pull request merged**